### DISCUSSION TITLE
Usings Order

### DISCUSSION THREAD

**Discussion by Morbidous** [MNES-Emulator/MNES#21] [General]
Maybe I should be using "general" and not "Q&A" oh well. (actually maybe I _should_ be using Q&A because I _am_ asking you a question... oh well.)

Do you have a preference for usings order?

Normally I order them by "importance". This is System, followed by Microsoft, JetBrains, other third-party libs, then internal core libs, then internal non-core libs, then libs that relate to this project/system.

Another option is to just put them in total alphabetical order, ignoring importance.

I don't know what the overall solution to this is; I think people use alphabetic. I always wish I could define a total order somewhere and then auto place them in that order. Any auto cleanup would usually use alphabetic I think but I don't like that alphabetic is not a logical order, it might as well be arbitrary. If I see things are alphabetic I might not change them but I'm going to feel like I want to put them in my order. I might do that but maybe not. You can let me know if you have a preference.

**Comment by jamieyello** [comment]
Alphabetic is how I've done it in the past, but I don't have any kind of tool that will do it automatically for me. I just put the "`using static Thing.Class;`" after all the other usings.

**Reply by Morbidous** [reply]
k I will aim for alphabetical (and yeah `using static` and aliases go after the regular ones)

(random thought: I wonder if aliases can go in global usings. I'm not entirely sold on global usings but if aliases can go there that would actually be quite handy in some cases)
